### PR TITLE
Django 1.10 support

### PR DESCRIPTION
**Pull request by zoidbergwill**
PR for tracking support for Django 1.10

- [x] `django.conf.urls.pattern` has been removed
- [x] checking namespace/app_name patterns stuff still works (I blindly deleted lines that might 
need to come back)
- [x] django rest framework issue:

```
Unhandled exception in thread started by <function check_errors.<locals>.wrapper at 0x102796158>
  File "/src/git/ctpug/wafer/wafer/users/urls.py", line 14, in <module>
    url(r'^api/', include(router.urls)),
  File "/src/git/ctpug/wafer/virtualenv/lib/python3.5/site-packages/rest_framework/routers.py", line 79, in urls
    self._urls = self.get_urls()
  File "/src/git/ctpug/wafer/virtualenv/lib/python3.5/site-packages/rest_framework/routers.py", line 321, in get_urls
    urls = format_suffix_patterns(urls)
  File "/src/git/ctpug/wafer/virtualenv/lib/python3.5/site-packages/rest_framework/urlpatterns.py", line 64, in format_suffix_patterns
    return apply_suffix_patterns(urlpatterns, suffix_pattern, suffix_required)
  File "/src/git/ctpug/wafer/virtualenv/lib/python3.5/site-packages/rest_framework/urlpatterns.py", line 27, in apply_suffix_patterns
    view = urlpattern._callback or urlpattern._callback_str
AttributeError: 'RegexURLPattern' object has no attribute '_callback'
```

### CODE REVIEW

**Comment by drnlm** [comment]
Django rest framework gained support for Django 1.10 with https://github.com/tomchristie/django-rest-framework/pull/4158 , but that hasn't made it into a release yet.

**Comment by drnlm** [comment]
Since we've dropped support for Django 1.7, and django-restframework has had a release supporting 1.10, we should resume looking at this.

**Comment by zoidbergwill** [comment]
Found a new PR to subscribe to, which is blocking this one: https://github.com/zsiciarz/django-markitup/pull/19

Though there is some tidying up I can still do for the namespacing stuff.

File: .travis.yml [hunk 1]
```diff
@@ -6,25 +6,45 @@ python:
...
+  directories:
+    - $HOME/.cache/pip
+
 before_install:
    - if [ "$TESTDB" = "postgres" ]; then pip install -q psycopg2 ; fi
 # command to install dependencies,
 install:
    # Install the right version of Django first
-   - pip install 'Django~='"$DJANGO_VERSION".0
+   - pip install $DJANGO
```
Comment (hodgestar) [inline .travis.yml]: If we're going to have complicated things in `$DJANGO`, it should probably be quoted here.

**Comment by zoidbergwill** [comment]
The one Travis run is trying to build CTPUG's django 1.10 branch instead of my fork's branch.

**Pull request merged**